model_wrappers.py

```diff
@@ -138,43 +96,215 @@ def get_judge_confidence(
                 proof_b=answer_b.proof,
                 response_b=response_b,
             )
-        full_prompt = self._format_judge_prompt(unformatted_prompt)
-        input_ids = self.tokenizer.encode(full_prompt, return_tensors="pt").to(
-            self.model.device
-        )
-        output = self.model(input_ids).logits[0, -1, :]
-        probs = output.softmax(dim=0)
-
-        correct_answer_prob = probs[self.tokenizer.encode(letters[0])[-1]].item()
-        incorrect_answer_prob = probs[self.tokenizer.encode(letters[1])[-1]].item()
-        return correct_answer_prob / (correct_answer_prob + incorrect_answer_prob)
+        if use_chat_completions is None:
+            use_chat_completions = hasattr(self, "PREFER_CHAT_COMPLETIONS") and self.PREFER_CHAT_COMPLETIONS
 
+        if use_chat_completions:
```

Comment:
Might be misreading but it looks like the chat-completions API would never allow the judge to have the open parenthesis "(" added as words_in_mouth, but that words_in_mouth addition is important for getting accurate results (otherwise the highest-probability tokens will be things other than 'A' and 'B')

Reply:
Never mind I'm wrong: https://huggingface.co/docs/transformers/en/main_classes/tokenizer#transformers.PreTrainedTokenizer.apply_chat_template